module/VuFind/tests/unit-tests/src/VuFindTest/Section/AbstractSectionTestCase.php

```diff
@@ -0,0 +1,247 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Section;
+
+use VuFind\Auth\ILSAuthenticator;
+use VuFind\Auth\Manager;
+use VuFind\Config\AccountCapabilities;
+use VuFind\Config\YamlReader;
+use VuFind\ILS\Connection;
+use VuFind\Navigation\AbstractMenu;
+use VuFind\Navigation\AccountMenu;
+use VuFind\Navigation\AdminMenu;
+use VuFind\Navigation\NavigationInterface;
+use VuFind\Navigation\PluginManager as NavigationManager;
+use VuFind\Section\PluginManager as SectionManager;
+use VuFind\Section\SectionInterface;
+use VuFind\Section\SectionService;
+use VuFind\Section\SectionServiceInterface;
+use VuFindTest\Container\MockContainer;
+use VuFindTest\Feature\ConfigRelatedServicesTrait;
+
+/**
+ * Abstract base class for navigation plugin tests.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+abstract class AbstractSectionTestCase extends \PHPUnit\Framework\TestCase
+{
+    use ConfigRelatedServicesTrait;
+
+    /**
+     * Get a container with section related services.
+     *
+     * @return MockContainer
+     */
+    protected function getContainerWithSectionRelatedServices(): MockContainer
+    {
+        $container = new MockContainer($this);
+        $this->addSectionRelatedServicesToContainer($container);
+        return $container;
+    }
+
+    /**
+     * Add section related services to a mock container.
+     *
+     * @param MockContainer $container Mock container
+     *
+     * @return void
+     */
+    protected function addSectionRelatedServicesToContainer(MockContainer $container)
+    {
+        $container->set(YamlReader::class, new YAMLReader($this->getPathResolver()));
+        $sectionManager = new SectionManager($container);
+        $container->set(SectionManager::class, $sectionManager);
+        $navigationManager = new NavigationManager($container);
+        $container->set(NavigationManager::class, $navigationManager);
+        $service = new SectionService(
+            $container->get(YamlReader::class),
+            $sectionManager,
+            $navigationManager,
+            'en',
+            ['en', 'fi']
+        );
+        $container->set(SectionServiceInterface::class, $service);
+        $this->getAccountMenu([], [], $container);
```

Comment:
There's something a bit confusing and circular about the fact that we use getAccountMenu and getAdminMenu to set up section plugins, but the setSectionPlugin method may optionally call this method. Would it make more sense to move all the plugin setup logic directly into this method, and require that we pass an appropriately configured container to getAccountMenu and getAdminMenu? I think that might simplify things and make the logic more easily comprehensible.